### DISCUSSION TITLE
Samsung G9 not working + solutions

### DISCUSSION THREAD

**Discussion by StErMi** [MonitorControl/MonitorControl#1580] [General]
**Checklist**

- [x] I have searched for existing issues
- [] I have looked through [the Monitor Troubleshooting wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [x] I have updated MonitorControl to the latest version

**Describe the issue**

I tried to make research but I really don't know if the Samsung G9 supports DDC.
I load up the application and try to change the brightness of it but it does not seem to work at all.

When I click on the app all the areas are greyed out, I don't know if it means that the monitor is not supported

**Expected behavior**

When I change the mbp brightness the external monitor should sync according to the same value.

**Screenshots**

None

**Additional context**

None

**Environment Information (please complete the following information):**
 - macOS version: 10.15.7
 - MonitorControl version: 2.1.0
 - Monitor(s): Samsung G9
 - Mac model: Macbook Pro 13" 2019
 - Monitor Cable(s)/Connection(s): USB-C to DisplayPort 1.4

**Comment by haakon-nergaard** [comment]
If it helps, I've just tested the Samsung G9 with ddcctl and that didn't work either.


    macOS version: 10.15.7
    MonitorControl version: 2.1.0
    Monitor(s): Samsung G9
    Monitor Cable(s)/Connection(s): miniDP -> DisplayPort

**Comment by douginator2000** [comment]
Same issue for me on the Samsung CRG9. The tool launches into the menu bar but the sliders are grayed out. Brightness up/down buttons show on-screen indication and it does adjust the slider for MonitorControl, but it does not actually change the brightness.

**Comment by jrfernandes** [comment]
I have a CRG9 and it can control the brightness, but the volume doesn't work. The sliders are functional, it just doesn't change the volume.

**Comment by douginator2000** [comment]
Did you have to do anything in particular to make it work @jrfernandes ?

I've found that if I switch freesync modes or turn the monitor off/on, the brightness buttons do work on my keyboard.

**Comment by waydabber** [comment]
Hi all, this seems to be a monitor issue and probably there is nothing much to do on MonitorControl's side especially if a different tool (namely ddcctl) can't control the display either. The new 3.0.0 version has a great software fallback option (better than nothing) if you can't make this work.

I'll close this issue for now but feel free to comment if there is anything we can help with! Thank you!

**Comment by voltechs** [comment]
Adding more information (searched for existing issues like a good citizen).

I have the [Samsung 49" Odyssey G9 Gaming Monitor](https://www.samsung.com/us/computing/monitors/gaming/49-odyssey-g9-gaming-monitor-lc49g97tssnxdc/?substitution=gaming) (LC49G95T). Maybe somewhat irrelevantly, I'm noticing that Samsung has two SKUs for this, one of which is discontinued. I purchased this a few days ago and it looks like they sent me a discontinued version so I'll have to get on the phone with them and see what's going on. That being said, I can confirm that even with the latest firmware (1015.0 at this time) no hardware dimming.

I "upgraded" from the [CRG9](https://www.samsung.com/us/computing/monitors/gaming/49-crg9-dual-qhd-curved-qled-gaming-monitor-lc49rg90ssnxza/) model, which did work great with MonitorControl dimming.

All that to say, is there anything I can do as an owner of the device to help you debug/develop support for it?

Edit: I decided to do some poking around, but I was quickly stymied. I was looking around on ddcctl, and found [this feature request](https://github.com/kfix/ddcctl/issues/93) (because I use PIP, but that's irrelevant) and saw mention of an app (for Windows) called ControlMyMonitor. I thought maybe I would get a bunch of control codes or whatnot from it to relay here in hopes that it would help further support for the monitor, but I only got this error instead (see screenshot). I tried turning PIP off too, for what it's worth. Seems either I have a lemon, or these monitors simply don't support DDC? Seems unlikely and if so, outrageous.
<details>
  <summary>Screenshot</summary>

  ![image](https://user-images.githubusercontent.com/1700514/154724111-056d0ee0-354a-434d-b70b-4022d55e4de4.png)
</details>

Edit2: More sleuthing, found [this documentation](https://www.ddcutil.com/monitor_notes/#samsung-lc49g95t-odyssey-g9) from www.ddcutil.com. 
> Two users have reported that DDC/CI does not work over DisplayPort. In both cases they were using the amdgpu video driver. This was true no matter which DisplayPort version was chosen in the OSD. As of 12/2020, this is not fixed by the most recent monitor firmware update.

**Comment by henryhobhouse** [comment]
I've got a CRG9 and have also been trying to get something working here. 

I'm almost there. I'm using monitor control to adjust brightness but as other have said no dice on volume. I have, however, found that eqMac does work adjusting volume including supporting native volume keys. 

The only downside is eqMac isn't particularly stable and crashes frequently when coming out of sleep or changing monitor setting (in my case turning pbp on or off)

So almost there but not quite. Not sure eqMac is doing a different path to make volume work but would be above awesome if monitor control could utilise there approach as it's brilliant, and importantly, stable app.

**Comment by voltechs** [comment]
> I've got a CRG9 and have also been trying to get something working here.

FYI, this issue/thread is for the CRG9's "successor" (of sorts), the Odyssey G9 Gaming Monitor, not the CRG9 itself. I also noted above that I upgraded from that monitor and MonitorControl was working just fine with it (as far as brightness). I think we're still SOL on the Odyssey G9 for now 😔

**Comment by TeaRex-coder** [comment]
Not being able to use DDC on DisplayPort with my G9 has been bothering me as well :\

I've been working on a [fix](https://x.com/BoutoilleAlex/status/1809245471795212388), so us G9 users might not be SOL

It's a custom PCB that uses the HDMI input to send the DDC commands from the host computer while using DP for video.

Still working out a few core features, but trying to gauge from the community if ppl actually care lol.

If this isn't a great place to talk about this, feel free to [email me](mailto:alexandre@boutoille.com) or DM me on [Twitter/X](https://twitter.alexandreboutoille.com)

**Comment by waydabber** [comment]
Wow, nice! :)

**Comment by voltechs** [comment]
That's pretty clever! I'm definitely interested. Keep us posted!

**Comment by TeaRex-coder** [comment]
Thanks, I'll keep you guys posted :)

**Comment by waydabber** [comment]
I'll move this over to discussion so it is not a closed issue (difficult to find).
